pilot/pkg/xds/endpoint_builder.go

```diff
@@ -413,7 +413,7 @@ func buildEnvoyLbEndpoint(b *EndpointBuilder, e *model.IstioEndpoint, mtlsEnable
 	}
 
 	// Otherwise has ambient enabled. Note: this is a synthetic label, not existing in the real Pod.
```

Comment:
What happens if it DOES have the label ?  I don't think we remove it - so it would work as intended ( would be treated as ambient ). For example if it's a VM or in some other environment - and it does implement HBone native.

Reply:
The "synthetic label" is actually not how its implemented anymore. But the _real_ label `networking.istio.io/tunnel=http` is supported. Ill update the comment